core/src/main/java/org/apache/iceberg/SchemaParser.java

```diff
@@ -141,11 +145,39 @@ static void toJson(Types.MapType map, JsonGenerator generator) throws IOExceptio
   }
 
   static void toJson(Type.PrimitiveType primitive, JsonGenerator generator) throws IOException {
-    generator.writeString(primitive.toString());
+    switch (primitive.typeId()) {
```

Comment:
Primitives are encoded as type strings. This method should not change and `GeometryType.toString` should produce a string that can be parsed by `Types.fromTypeName`.

Reply:
Geography and Geometry types are encoded as JSON objects according to https://iceberg.apache.org/spec/#schemas, we can update the spec if strings are always preferred for primitive types including geospatial types.

Reply:
I missed that in the spec.

@szehon-ho, why are we using objects for these when we also support parsing them as we do for `decimal` and `fixed`? Should we update the spec?

Reply:
Im ok with that, initially I thought its a bit complex as the crs can be projjson content.  But now it is a table property, so should work.